shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -404,7 +404,21 @@ public PlatformViewRegistry getRegistry() {
     return registry;
   }
 
-  public void onFlutterViewDestroyed() {
+  /**
+   * Invoked when the {@link io.flutter.embedding.engine.FlutterEngine} that owns this {@link
+   * PlatformViewsController} attaches to JNI.
+   */
+  public void onAttachedToJNI() {
+    // Currently no action needs to be taken after JNI attachment.
+  }
+
+  /**
+   * Invoked when the {@link io.flutter.embedding.engine.FlutterEngine} that owns this {@link
+   * PlatformViewsController} detaches from JNI.
+   */
+  public void onDetachedFromJNI() {
```

Comment:
In keeping with how Android/iOS tend to name these things, this could be onAttachedToJNI and onDetachingFromJNI since strictly at the time of this call, JNI is not 'detached' yet.

Reply:
The name `onDetachedFromJNI` was taken from the existing methods in `DartExecutor`
(https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java#L91)

Reply:
Doh you're right, that's already there. Though since there's only one of it in the entire flutter codebase and no other references in g3, I'd rename that one in DartExecutor too rather than keeping this name with the other one.

Reply:
I would actually suggest that we leave it. Names that imply ongoing operations are impossible to reason about. I think we're better off suggesting to clients that we are disconnected, rather than "disconnecting".

Reply:
The confusion would be if the client doesn't know whether it's disconnecting or disconnected, it shouldn't call any of the downstream methods like flushAllViews etc which uses JNI

Reply:
The questions that will arise (speaking as someone who has asked these same questions for those unusual Android methods that do this) are: what does it mean to be "disconnecting"? How many operations happen while "disconnecting", 2, 5, 10? Where in the order of "disconnection" events am I being called? Which pieces of the subsystem are still available to me and which are gone?

On the other hand, saying that you are "disconnected" means one very specific thing. All the stuff you are referencing, get rid of it. You can't call anything, anymore. It may not be the exact truth, because maybe you could call 1 or 2 things, but it's a very safe and concrete lie, as opposed to an unstable and unanswerable truth.

Reply:
BTW, if you want a concrete example in Android that irritates people a lot, checkout `onPostResume()`. And that's not even as ambiguous as "onDisconnecting()".